src/analysis/convex/independent.lean

```diff
@@ -0,0 +1,226 @@
+⟨λ hc, hc.comp_embedding
+  (⟨λ i, ⟨p i, set.mem_range_self _⟩, λ x y h, hi (subtype.mk_eq_mk.1 h)⟩ : ι ↪ set.range p),
+  convex_independent.range⟩
+
+/-- If a family is convex independent, a point in the family is in the convex hull of some of the
+points given by a subset of the index type if and only if the point's index is in this subset. -/
+@[simp] protected lemma convex_independent.mem_convex_hull_iff {p : ι → E}
+    (hc : convex_independent p) (s : set ι) (i : ι) :
+  p i ∈ convex_hull (p '' s) ↔ i ∈ s :=
+⟨hc _ _, λ hi, subset_convex_hull  _ (set.mem_image_of_mem p hi)⟩
+
+/-- If a family is convex independent, a point in the family is not in the convex hull of the other points. See `convex_independent_set_iff_not_mem_convex_hull_diff` for the `set` version.  -/
+lemma convex_independent_iff_not_mem_convex_hull_diff {p : ι → E} :
+  convex_independent p ↔ ∀ i s, p i ∉ convex_hull (p '' (s \ {i})) :=
+begin
+  refine ⟨λ hc i s h, _, λ h s i hi, _⟩,
+  { rw hc.mem_convex_hull_iff at h,
+    exact h.2 (set.mem_singleton _) },
+  { by_contra H,
+    refine h i s _,
+    rw set.diff_singleton_eq_self H,
+    exact hi }
+end
+
+lemma convex_independent_set_iff_inter_convex_hull_subset {s : set E} :
+  convex_independent (λ x, x : s → E) ↔ ∀ t, t ⊆ s → s ∩ convex_hull t ⊆ t :=
+begin
+  split,
+  { rintro hc t h x ⟨hxs, hxt⟩,
+    refine hc {x | ↑x ∈ t} ⟨x, hxs⟩ _,
+    rw subtype.coe_image_of_subset h,
+    exact hxt },
+  { intros hc t x h,
+    rw ←subtype.coe_injective.mem_set_image,
+    exact hc (t.image coe) (subtype.coe_image_subset s t) ⟨x.prop, h⟩ }
+end
+
+/-- If a set is convex independent, a point in the set is not in the convex hull of the other
+points. See `convex_independent_iff_not_mem_convex_hull_diff` for the indexed family version.  -/
+lemma convex_independent_set_iff_not_mem_convex_hull_diff {s : set E} :
+  convex_independent (λ x, x : s → E) ↔ ∀ x ∈ s, x ∉ convex_hull (s \ {x}) :=
+begin
+  rw convex_independent_set_iff_inter_convex_hull_subset,
+  split,
+  { rintro hs x hxs hx,
+    exact (hs _ (set.diff_subset _ _) ⟨hxs, hx⟩).2 (set.mem_singleton _) },
+  { rintro hs t ht x ⟨hxs, hxt⟩,
+    by_contra h,
+    exact hs _ hxs (convex_hull_mono (set.subset_diff_singleton ht h) hxt) }
+end
+
+--TODO: Finish
+-- lemma affine_independent.convex_independent {p : ι → E} (ha : affine_independent ℝ p) :
+--   convex_independent p :=
+-- begin
+--   rw convex_independent_iff_finset,
+--   intros s x hx,
+--   rw finset.convex_hull_eq at hx,
+--   obtain ⟨w, _, hw, hx⟩ := hx,
+--   rw ←finset.mem_coe,
+--   suffices h : (s : set ι).indicator (w ∘ p) x ≠ 0,
+--   { exact set.mem_of_indicator_ne_zero h },
+--   --rw ←ha.injective.mem_finset_image_iff,
+--   rw finset.sum_image (ha.injective.inj_on _) at hw,
+--   rw [ha.indicator_eq_of_affine_combination_eq s {x} (w ∘ p) 1 hw finset.sum_singleton _,
+--     finset.coe_singleton, set.indicator_of_mem (set.mem_singleton _)],
+--   exact one_ne_zero,
+--   sorry,
+-- end
```

Comment:
I'm not sure if there is a consensus position on incomplete proofs, but I'd request that you add a few lines saying what the obstacle to finishing it is.

Reply:
It's in the module docstring. I will copy it down there.